lib/docker.js

```diff
@@ -0,0 +1,518 @@
+            console.log(opts.commit_msg);
+            return;
+        }
+        return chained_pgit([
+            // send them for review
+            ['review', '-R'],
+            // get back to master
+            ['checkout', 'master']
+        ]).then(function() {
+            console.log("\n\nChanges sent to Gerrit for review!");
+        });
+    });
+
+}
+
+
+/**
+ * Determines the UID and GID to run under in the container
+ *
+ * @return {Promise} a promise resolving when the check is done
+ */
+function get_uid() {
+
+    if(opts.deploy) {
+        // get the deploy repo location
+        return promised_spawn(['git', 'config', 'deploy.dir'], true)
+        .then(function(dir) {
+            if(!dir) {
+                console.error('ERROR: You must set the location of the deploy repo!');
+                console.error('ERROR: Use git config deploy.dir /full/path/to/deploy/dir');
+                process.exit(2);
+            }
+            opts.dir = dir;
+            // make sure that the dir exists and it is a git repo
+            return fs.statAsync(dir + '/.git');
+        }).then(function(stat) {
+            opts.uid = stat.uid;
+            opts.gid = stat.gid;
+        }).catch(function(err) {
+            console.error('ERROR: The deploy repo dir ' + opts.dir + ' does not exist or is not a git repo!');
+            process.exit(3);
+        });
+    }
+
+    // get the uid/gid from statting package.json
+    return fs.statAsync('package.json')
+    .then(function(stat) {
+        opts.uid = stat.uid;
+        opts.gid = stat.gid;
+    }).catch(function(err) {
+        console.error('ERROR: package.json does not exist!');
+        process.exit(4);
+    });
+
+}
+
+
+/**
+ * Main process signal handler
+ */
+function sig_handle() {
+    if(child) {
+        child.kill('SIGINT');
+    }
+}
+
+
+function main(options, configuration) {
+
+    // set up working options
+    if(options.hasOwnProperty('forceBuild')) {
+        // called from service-runner, translate
+        opts = {
+            tests: options.dockerTest,
+            coverage: false,
+            deploy: options.build,
+            verbose: false,
+            need_build: options.forceBuild,
+            review: options.sendReview,
+            path: options.basePath
+        };
+    } else {
+        // called from the command line
+        opts = options;
+    }
+
+    // config and package info
+    config = configuration;
+    if(!config) {
+        config = yaml.safeLoad(fs.readFileSync(path.join(opts.path, 'config.yaml')));
+    }
+    pkg = config.package;
+    if(!pkg) {
+        pkg = require(path.join(opts.path, 'package.json'));
+    }
+
+    // target info
+    try {
+        targets = yaml.safeLoad(fs.readFileSync(path.join(opts.path, 'targets.yaml')));
+    } catch(e) {
+        // no such file or wrong format,
+        // set the defaults
+        targets = {
+            debian: 'buildpack-deps:jessie',
```

Comment:
Should we use https://registry.hub.docker.com/_/debian/ ? It is [recommended on the Debian wiki](https://wiki.debian.org/Docker). They also have [a review](https://joeyh.name/blog/entry/docker_run_debian/) of the differences between the image and a vanilla debootstrap that sounds reasonable.